### PR TITLE
Adds ability Embody Aspect + minor fix to Hospitality

### LINKED ISSUE #3815: ABILITY_EMBODY_ASPECT
### Description

Title (and variants) are missing functionality

### Version

upcoming (Edge)

### Upcoming Version

_No response_

### Discord contact info

_No response_

### LINKED ISSUE #3820: Neutralizing Gas / Intrepid Sword interaction causes a game loop / freeze
### Description

When you switch out a mon with Neutralizing Gas, Intrepid Sword is activated as it should but then the ability user is switched out into a bad egg. I haven't tested it in-game but I imagine the same interaction will happen. 

https://github.com/rh-hideout/pokeemerald-expansion/assets/93446519/895195e0-9628-408a-b7a0-9357d704889e

Test in #3821

### Version

1.7.0 (Default)

### Upcoming Version

_No response_

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Closes #3815
Fixes #3820

Fixed a potential bug with Hospitality with Dynamax. Based on Heal Pulse interaction with dynamax, the amount recovered is based on original HP but the dynamax health is checked.

### CODE REVIEW

File: src/data/trainer_parties.h [hunk 1]
```diff
@@ -3207,10 +3207,10 @@ static const struct TrainerMon sParty_Drake[] = {
 static const struct TrainerMon sParty_Roxanne1[] = {
     {
     .iv = TRAINER_PARTY_IVS(12, 12, 12, 12, 12, 12),
-    .lvl = 12,
+    .lvl = 100,
     .species = SPECIES_GEODUDE,
     .heldItem = ITEM_NONE,
-    .moves = {MOVE_TACKLE, MOVE_DEFENSE_CURL, MOVE_ROCK_THROW, MOVE_ROCK_TOMB}
+    .moves = {MOVE_PURSUIT}
```
Comment (Bassoonian) [inline src/data/trainer_parties.h:3213]: please revert this :)

File: src/data/trainer_parties.h [hunk 2]
```diff
@@ -3219,13 +3219,6 @@ static const struct TrainerMon sParty_Roxanne1[] = {
...
     .moves = {MOVE_TACKLE, MOVE_DEFENSE_CURL, MOVE_ROCK_THROW, MOVE_ROCK_TOMB}
     },
-    {
-    .iv = TRAINER_PARTY_IVS(24, 24, 24, 24, 24, 24),
-    .lvl = 15,
-    .species = SPECIES_NOSEPASS,
-    .heldItem = ITEM_ORAN_BERRY,
-    .moves = {MOVE_BLOCK, MOVE_HARDEN, MOVE_TACKLE, MOVE_ROCK_TOMB}
-    }
 };
```
Comment (Bassoonian) [inline src/data/trainer_parties.h:3222]: please revert this also

File: src/data/trainers.h [hunk 3]
```diff
@@ -3187,7 +3187,7 @@ const struct Trainer gTrainers[] = {
         .trainerPic = TRAINER_PIC_LEADER_ROXANNE,
         .trainerName = _("ROXANNE"),
         .items = {ITEM_POTION, ITEM_POTION, ITEM_NONE, ITEM_NONE},
-        .doubleBattle = FALSE,
+        .doubleBattle = TRUE,
```
Comment (Bassoonian) [inline src/data/trainers.h:3190]: and this

**Review by Bassoonian — CHANGES_REQUESTED** [review]
You left some debugging stuff in :)

**Comment by AlexOn1ine** [comment]
Removed leftover stuff.

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**